dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/get-fixed-fee-for-facility.ts

```diff
@@ -1,17 +1,19 @@
 import { ReportPeriod } from '@ukef/dtfs2-common';
-import { calculateUkefShareOfUtilisation, getLatestTfmFacilityValues } from '../../../../../helpers';
+import { calculateUkefShareOfUtilisation } from '../../../../../helpers';
 import { calculateFixedFee } from './calculate-fixed-fee';
+import { LatestTfmFacilityValues } from '../../../../../types/tfm/tfm-facility';
 
 /**
  * Gets the fixed fee for the given report period
  * calculates ukef share of utilisation from provided utilisation value and cover percentage
  * @param facilityId - The (ukef) facility id
  * @param utilisation - The facility utilisation
  * @param reportPeriod - The report period
+ * @param tfmFacilityValues - TFM facility values
  * @returns The fixed fee for the supplied report period
  */
-export const getFixedFeeForFacility = async (facilityId: string, utilisation: number, reportPeriod: ReportPeriod) => {
-  const { coverEndDate, dayCountBasis, interestPercentage, coverPercentage } = await getLatestTfmFacilityValues(facilityId, reportPeriod);
+export const getFixedFeeForFacility = (utilisation: number, reportPeriod: ReportPeriod, tfmFacilityValues: LatestTfmFacilityValues) => {
+  const { coverPercentage, coverEndDate, interestPercentage, dayCountBasis } = tfmFacilityValues;
```

Comment:
How about passing in the params specifically instead of needing `tfmFacilityValues` and `LatestTfmFacilityValues`? The current param/type feels ambiguous to me - lots more "values" available in a facility; Specific params would be clearer IMO

```ts
coverPercentage, coverEndDate, interestPercentage, dayCountBasis
```

Reply:
changed